app/models/report.rb

```diff
@@ -14,4 +21,16 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def create_mentions
+    mentioned_report_ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.uniq
+    mentioned_report_ids.map(&:to_i).each do |id|
+      active_mentions.create!(mentioned_report_id: id, mentioning_report_id: self.id)
```

Comment:
```suggestion
      active_mentions.create!(mentioned_report_id: id)
```

だけでもうまく動きませんかね？ （未検証）

Reply:
ありがとうございます！
動きました！指定しないとデフォルトで`self.id`になるんですね。
`foreign_key`のおかげっぽいということはわかったのですが、関連付けのガイドを読んでもいまいち理解まではできませんでした💦

Reply:
`foreign_key`というより、`has_many`かな？
`has_many` で適切に関連付けしておくと、 `blog.comments.create(content: 'しっかりやりな')` みたいなコードを書いたときにRailsの魔力で `comment.blog_id` に `blog.id` が自動的に設定されます。

もちろん、今回は少し特殊な関連付けなので、`foreign_key`も正しく指定する必要がありますが。
ただ、僕の中では「`foreign_key`が正しければ」というよりも、「`has_many`の設定が正しければ」と説明する方がしっくりきます。

> 関連付けのガイドを読んでもいまいち理解まではできませんでした💦

https://railsguides.jp/association_basics.html でいうと、

> 上のように関連付けを追加したことで、特定の著者の新しい書籍を1冊追加する作業が以下のように1行で書けるようになりました。

```ruby
@book = @author.books.create(published_at: Time.now)
```

のあたりが該当しそうですが、Rails初心者はこの説明だけだと「？？？」になりそうですね😅

Reply:
丁寧なご説明ありがとうございます！
正直「？？？」となっております😅
でもせっかくなので、いただいたご説明を基にもう少し調べて日報にまとめてみようと思います！